control/controlclient/direct.go

```diff
@@ -1539,6 +1503,9 @@ func answerC2NPing(logf logger.Logf, c2nHandler http.Handler, c *http.Client, pr
 }
 
 func sleepAsRequested(ctx context.Context, logf logger.Logf, timeoutReset chan<- struct{}, d time.Duration, clock tstime.Clock) error {
+	if timeoutReset == nil {
+		timeoutReset = make(chan struct{})
+	}
```

Comment:
this is confusing, whats the point of making a channel here that no one can read from and always blocks anyway?

Reply:
Oh, I was wrong about nil writable channels in select. Will remove. (They're fine: https://go.dev/play/p/po5IcntgwFU)

Reply:
yeah, they just block forever